parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -171,8 +191,8 @@ private void startRowGroup() {
     this.nextCheckRecordCount = Math.min(Math.max(recordCount / 2, 100), 10000);
     this.recordCount = 0;
 
-    PageWriteStore pageStore = pageStoreCtor.newInstance(
-        compressor, parquetSchema, props.getAllocator());
+    PageWriteStore pageStore = pageStoreCtorParquet.newInstance(
+          compressor, parquetSchema, props.getAllocator(), Integer.MAX_VALUE);
```

Comment:
This will fail in case of `1.10` as it does not expect the last `int`, isn't it?

Also, why not pass `ParquetProperties.DEFAULT_COLUMN_INDEX_TRUNCATE_LENGTH` instead of `Integer.MAX_VALUE`? It looks like `Integer.MAX_VALUE` is used in test cases only (for this particular value).

Reply:
Thanks for the explanation, the update looks cool!